central/auth/service/service_impl.go

```diff
@@ -9,10 +9,20 @@ import (
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/pkg/auth/user"
 	"github.com/stackrox/rox/pkg/grpc/authn"
-	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	authzUser "github.com/stackrox/rox/pkg/grpc/authz/user"
 	"google.golang.org/grpc"
 )
 
+var (
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		authzUser.Authenticated(): {
```

Comment:
Check here limits who can access this endpoint to non-service identities(because `id.Permissions()` will return empty and that will lead to 401 [code](https://github.com/stackrox/stackrox/blob/4bd4a3c6c4455d3425f17d165f105c8bef6a412f/pkg/grpc/authz/user/user.go#L43-L45)). Even though I couldn't find references to service identities actually using this endpoint, there is a [code supporting that case](https://github.com/stackrox/stackrox/blob/4bd4a3c6c4455d3425f17d165f105c8bef6a412f/central/auth/service/service_impl.go#L81-L83). It would be nice to figure out if we can then stop supporting service identities in this endpoint - if not, then using just `user.Authenticated()` is not enough here

Reply:
Good catch, Ivan!

I'd say let's not question now whether we can drop the service identity code branch. There is no harm in having this endpoint available for services, which IIRC mean stackrox services only [1] because of the cert validation that is a prerequisite for successful service identity construction.

[1] https://github.com/stackrox/stackrox/blob/66b99b741b396f9ce3e4b4e9475571b2607d2831/proto/storage/service_identity.proto#L21-L35